File: Common/InMotionGIT.Common - Copy/Helpers/SQL.vb
Namespace Helpers

    Public Class SQL

        Public Shared Function SafeSqlLikeClauseLiteral(ByVal inputSQL As String) As String
            Dim s As String = inputSQL
            s = inputSQL.Replace("'", "''")
            s = s.Replace("[", "[[]")
            s = s.Replace("%", "[%]")
            s = s.Replace("_", "[_]")
            Return s
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/SQL.cs

namespace InMotionGIT.Common.Helpers
{

    public class SQL
    {

        public static string SafeSqlLikeClauseLiteral(string inputSQL)
        {
            string s = inputSQL;
            s = inputSQL.Replace("'", "''");
            s = s.Replace("[", "[[]");
            s = s.Replace("%", "[%]");
            s = s.Replace("_", "[_]");
            return s;
        }

    }

}